components/Login/index.js

```diff
@@ -0,0 +1,115 @@
+import { signIn, signOut, useSession } from "next-auth/react";
+import styled from "styled-components";
+import Image from "next/image";
+import { useState } from "react";
+
+export default function Login({ isHomepage }) {
+  const { data: session } = useSession();
+  const [isHovered, setIsHovered] = useState(false);
+
+  function handleSignIn() {
+    signIn("github", { callbackUrl: "/exercises" });
+  }
+
+  const ButtonComponent = isHomepage ? HomepageButton : AnimatedButton;
+
+  return (
+    <ButtonComponent
```

Comment:
There are some improvements I can propose to your code : 

1. Refactor the ButtonComponent as a reusable component: It's better to define the button in a separate file, making it easier to reuse and maintain.